ext/spl/spl_directory.c

```diff
@@ -2583,7 +2583,8 @@ PHP_METHOD(SplFileObject, fscanf)
 	int result = php_sscanf_internal(ZSTR_VAL(intern->u.file.current_line), ZSTR_VAL(format_str), (int)num_varargs, varargs, 0, return_value);
 
 	if (SCAN_ERROR_WRONG_PARAM_COUNT == result) {
-		WRONG_PARAM_COUNT;
+		zend_wrong_param_count();
+		RETURN_THROWS();
 	}
```

Comment:
For this and the ones in ext/standard, it feels providing a better error message due to format string issues would be great, but for the time being this is okay as it is the existing status quo